README.md

```diff
@@ -1,74 +1,168 @@
-The easiest way to get started is to run the following command. This will pull down any necessary dependencies that your app will require. You can think of this command as something incredibly similar to `bundle install` in Rails. 
+## Initial Setup
 
-`npm install`
+- Clone down repo using `git clone git@github.com:johnktravers/play.git`
+- Change into project directory
+- Run `npm install`
+- Create development database
+```
+psql
+CREATE DATABASE PUT_DATABASE_NAME_HERE_dev
+\q
+```
+- Create test database
+```
+psql
+CREATE DATABASE PUT_DATABASE_NAME_HERE_test
+\q
+```
+- Run migrations for development database using `knex migrate:latest`
+- Run migrations for test database using `knex migrate:latest --env test`
 
-#### Set up your local database
-You’ll need to figure out a name for your database. We suggest calling it something like `sweater_weather_dev`.  
+### Environment Variables
 
-To get things set up, you’ll need to access your Postgres instance by typing in `psql` into your terminal. Once there, you can create your database by running the comment `CREATE DATABASE PUT_DATABASE_NAME_HERE_dev;`. 
+The following environment variables are required.
+- **DATABASE_URL** set to the location of your production database
+- **MUSIXMATCH_API_KEY** set to your registered MusixMatch Developer API Key
 
-Now you have a database for your new project.
+## How to Use / Endpoints
 
-#### Migrations
-Once you have your database setup, you’ll need to run some migrations (if you have any). You can do this by running the following command: 
+### Favorites Endpoint
 
-`knex migrate:latest`
+#### 1) Getting a list of all favorite tracks
 
+When a user sends a `GET` request to `api/v1/favorites` it returns all favorited songs currently in the database.
 
-Instructions to create database, run migrations, and seed: 
+Response body will look like:
 ```
-psql
-CREATE DATABASE DATABASE_NAME_dev;
-\q
+[
+  {
+    "id": 1,
+    "title": "We Will Rock You",
+    "artistName": "Queen"
+    "genre": "Rock",
+    "rating": 88
+  },
+  {
+    "id": 2,
+    "title": "Careless Whisper",
+    "artistName": "George Michael"
+    "genre": "Pop",
+    "rating": 93
+  },
+]
+```
+
+#### 2) Getting a favorite track by id
+
+A user can send a `GET` request to `/api/v1/favorites/:id` and is returned the information for a single favorite.
 
-knex migrate:latest
-knex seed:run
+The response body will look like:
 ```
+  {
+    "id": 1,
+    "title": "We Will Rock You",
+    "artistName": "Queen"
+    "genre": "Rock",
+    "rating": 88
+  }
+```
+
+#### 3) Adding a favorite track
 
-#### Set up your test database
-Most of the setup is going to be same as the one you did before. You’ll notice one small difference with setting the environment flag to `test`.  
+A user can send a `POST` request to `/api/v1/favorites` that allows them to add a favorite song to the database.
 
+The user request payload will be:
+```
+{ title: "We Will Rock You", artistName: "Queen" }
```

Comment:
We might want to make these a little more general, like:

The user's request body must have the following format: `{ title: <song_title>, artist: <artist_name> }`.